slm_porosity/heat_transfer.f90

```diff
@@ -1114,6 +1140,35 @@ PURE FUNCTION laser_distribution (x_, nlocal, factor)
     laser_distribution = EXP(-SUM(factor_*(x_-x_center)**2, 2))/pi**(.5*(dim-1))
   END FUNCTION laser_distribution
 
+  FUNCTION laser_heat_source (x_, nlocal)
+    IMPLICIT NONE
+    REAL(pr), INTENT(IN) :: x_(nlocal,dim)
+    INTEGER,  INTENT(IN) :: nlocal
+    REAL(pr) :: laser_heat_source(nlocal)
+    REAL(pr) :: position(3), halfwidth(3)
+    INTEGER  :: i
+
+    position = initial_laser_position
+    position(1) = position(1) - scanning_speed*t
```

Comment:
There is `laser_position` function for this.

Reply:
`laser_position` have `dim` size. Here I have scalars, after implementation of laser strategy the question could be arises again.

Reply:
Your position also have `dim` size. Each PR should be independent as possible. Never rely on what you will fix sometime later. It is the deepest root of evil.

Reply:
Ok I would try to implement everything in `laser_position` and `laser_distribution`.

Reply:
However `laser_distribution` didn't have halfwidth parameter and there is no normalising on a factor_ in here.

Reply:
Therefore, think how do it in the best way.